### PR TITLE
Brand new TAF

### PR DESCRIPTION
**Pull request by OlegChub**

### CODE REVIEW

File: src/test/resources/config/credentials.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NjU0ODYzMzEsInVzZXJfbmFtZSI6InN1cGVyYWRtaW4iLCJhdXRob3JpdGllcyI6WyJST0xFX0FETUlOSVNUUkFUT1IiXSwianRpIjoiMTM0OWZhNTgtYzVkMC00MjgwLWE5MjAtZDYxNTllNmMzMzNmIiwiY2xpZW50X2lkIjoidWkiLCJzY29wZSI6WyJ1aSJdfQ.xH02au7d8AEeDkDGwwvuo9kjJ5U_CRdGrJiuddImkpw
```
Comment (dKatechev) [inline src/test/resources/config/credentials.properties]: please remove value

File: src/test/java/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public class BaseTest {
+    private static final Logger logger = LogManager.getLogger();
+
+    @BeforeEach
+    protected void setUp() {
+        logger.info("Test is starting ...");
+    }
+
+    @AfterEach
+    protected void stopBrowser() {
```
Comment (dKatechev) [inline src/test/java/BaseTest.java]: stop browser?

File: src/main/java/controllers/DashboardController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public ValidatableResponse getDashboards() {
+        return client.execute(new GetDashboard());
+    }
+
+    public String getDashboardName() {
+        Dashboards dashboards = ResponseHandler.getResponseAsClass(Dashboards.class,
+                getDashboards().extract().response());
+        return dashboards.getContent().get(0).getName();
```
Comment (dKatechev) [inline src/main/java/controllers/DashboardController.java]: Why get(0)? Why not get(1)?
Maybe better get dasboards and send them to tests? Or create method
```boolean hasDashboardWithName(String dashboardName)```?

File: src/main/java/api/request/GetDashboard.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package api.request;
+
+import api.BaseRequest;
+
+import static helpers.RequestBuilder.urlBuilder;
+
+public class GetDashboard extends BaseRequest {
+    {
+        method = "GET";
+        url = urlBuilder("/dashboard");
```
Comment (dKatechev) [inline src/main/java/api/request/GetDashboard.java]: url = buildUrl(...

File: src/main/java/models/Dashboards.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Dashboards {
+        public List<Content> content;
+    }
```
Comment (dKatechev) [inline src/main/java/models/Dashboards.java]: format?
please format everywhere

File: src/main/java/helpers/RequestBuilder.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+public class RequestBuilder {
+    private final static String PROJECT_NAME = "superadmin_personal";
+    private final static String API_VERSION = "api/v1/";
+
+    public static String urlBuilder(String endpoint) {
+        return getApiVersion() + getProjectName() + endpoint;
+    }
+
+    public static String getApiVersion() {
```
Comment (dKatechev) [inline src/main/java/helpers/RequestBuilder.java]: Used in this class only. No need to create getter.

File: src/main/java/helpers/RequestBuilder.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        return API_VERSION;
+    }
+
+    public static String getProjectName() {
```
Comment (dKatechev) [inline src/main/java/helpers/RequestBuilder.java]: Used in this class only. No need to create getter.

File: src/main/java/config/ProjectConfig.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import org.aeonbits.owner.Config;
+
+@Config.LoadPolicy(Config.LoadType.MERGE)
+@Config.Sources({
+        "system:properties",
+        "classpath:config/project.properties"
+})
+public interface ProjectConfig extends Config {
+    String host();
+    String projectName();
```
Comment (dKatechev) [inline src/main/java/config/ProjectConfig.java]: Not used.

File: src/main/java/api/request/GetDashboard.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package api.request;
+
+import api.BaseRequest;
+
+import static helpers.RequestBuilder.urlBuilder;
+
+public class GetDashboard extends BaseRequest {
+    {
+        method = "GET";
```
Comment (dKatechev) [inline src/main/java/api/request/GetDashboard.java]: method = org.apache.http.client.methods.HttpGet.METHOD_NAME;

File: src/main/java/api/ApiClient.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+public class ApiClient {
+
+    public ValidatableResponse execute(BaseRequest request) {
+        addDefaultHeaders(request);
+        authorize(request);
+        return request.execute();
+    }
+
+    private void addDefaultHeaders(BaseRequest request) {
+        request.headers.put("Content-Type", "application/json");
```
Comment (dKatechev) [inline src/main/java/api/ApiClient.java]: Please, do not use direct access to the fields.

File: src/main/java/api/BaseRequest.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+import static helpers.ConfigSetUp.getHost;
+import static io.restassured.RestAssured.given;
+
+public class BaseRequest {
+    protected String method;
+    protected String url;
+
+    protected Map<String, String> headers = new HashMap<>();
+    protected Map<String, String> pathParams = new HashMap<>();
+    protected Map<String, String> queryParams = new HashMap<>();
```
Comment (dKatechev) [inline src/main/java/api/BaseRequest.java]: headers, pathParams, queryParams protected?

File: src/main/java/controllers/DashboardController.java (hunk 3 above)
Comment (OlegChub) [reply src/main/java/controllers/DashboardController.java]: refactor methods

File: src/test/java/BaseTest.java (hunk 2 above)
Comment (OlegChub) [reply src/test/java/BaseTest.java]: typo. Renamed

File: src/test/resources/config/credentials.properties (hunk 1 above)
Comment (OlegChub) [reply src/test/resources/config/credentials.properties]: removed

File: src/main/java/api/request/GetDashboard.java (hunk 4 above)
Comment (OlegChub) [reply src/main/java/api/request/GetDashboard.java]: changed

File: src/main/java/models/Dashboards.java (hunk 5 above)
Comment (OlegChub) [reply src/main/java/models/Dashboards.java]: done

File: src/main/java/helpers/RequestBuilder.java (hunk 6 above)
Comment (OlegChub) [reply src/main/java/helpers/RequestBuilder.java]: removed

File: src/main/java/helpers/RequestBuilder.java (hunk 7 above)
Comment (OlegChub) [reply src/main/java/helpers/RequestBuilder.java]: removed

File: src/main/java/config/ProjectConfig.java (hunk 8 above)
Comment (OlegChub) [reply src/main/java/config/ProjectConfig.java]: removed

File: src/main/java/api/request/GetDashboard.java (hunk 9 above)
Comment (OlegChub) [reply src/main/java/api/request/GetDashboard.java]: replaced

File: src/main/java/api/ApiClient.java (hunk 10 above)
Comment (OlegChub) [reply src/main/java/api/ApiClient.java]: implemented getters

File: src/main/java/api/BaseRequest.java (hunk 11 above)
Comment (OlegChub) [reply src/main/java/api/BaseRequest.java]: made them private

File: src/main/java/controllers/DashboardController.java (hunk 3 above)
Comment (OlegChub) [reply src/main/java/controllers/DashboardController.java]: plus added hamcrest library and remade checking in the test

**Pull request merged**